packages/awsui-react-ts-website/sample/src/Auth.tsx

```diff
@@ -0,0 +1,121 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+
+import React, { createContext, useCallback, useEffect, useState, useMemo } from "react";
+import { Amplify } from "aws-amplify";
+import { Authenticator, ThemeProvider, Theme, useTheme } from "@aws-amplify/ui-react";
+
+/**
+ * Context for storing the runtimeContext.
+ */
+export const RuntimeConfigContext = createContext<any>({});
+
+/**
+   * Sets up the runtimeContext and Cognito auth.
+   * 
+   * This assumes a runtime-config.json file is present at '/'. In order for Auth to be set up automatically,
+   * the runtime-config.json must have the following properties configured: [region, userPoolId, userPoolWebClientId, identityPoolId].
+   */
+const Auth: React.FC<any> = ({children}) => {
+  const [runtimeContext, setRuntimeContext] = useState<any>({});
+  const [runtimeContextLoaded, setRuntimeContextLoaded] = useState(false);
+  const { tokens } = useTheme();
+  
+  // Customize your login theme
+  const theme: Theme = useMemo(() => ({
+    name: "AuthTheme",
+    tokens: {
+      components: {
+        passwordfield: {
+          button: {
+            _hover: {
+              backgroundColor: {
+                value: "white"
+              },
+              borderColor: {
+                value: tokens.colors.blue['40'].value
+              }
+            }
+          }
+        }
+      },
+      colors: {
+        background: {
+          primary: {
+            value: tokens.colors.neutral['20'].value,
+          },
+          secondary: {
+            value: tokens.colors.neutral['100'].value,
+          },
+        },
+        brand: {
+          primary: {
+            '10': tokens.colors.blue['20'],
+            '80': tokens.colors.blue['40'],
+            '90': tokens.colors.blue['40'],
+            '100': tokens.colors.blue['40'],
+          },
+        },
+      },
+    }
+  }), [tokens]);
+
+  useEffect(() => {
+    fetch("/runtime-config.json")
+      .then(response => response.json())
+      .then(runtimeContext => {
+        setRuntimeContext(runtimeContext);
+
+        runtimeContext.region &&
+        runtimeContext.userPoolId &&
+        runtimeContext.userPoolWebClientId &&
+        runtimeContext.identityPoolId &&
+        Amplify.configure({
+          Auth: {
+            region: runtimeContext.region,
+            userPoolId: runtimeContext.userPoolId,
+            userPoolWebClientId: runtimeContext.userPoolWebClientId,
+            identityPoolId: runtimeContext.identityPoolId,
+          }
+        });
+      })
+      .catch((e) => console.log(e))
+      .finally(() => setRuntimeContextLoaded(true));
```

Comment:
Might be a bit cleaner to use async/await with try/catch instead of `.then`/`.catch` etc?

Reply:
I tried that, however async methods are not supported with useEffect. The only way to do it would be to create a seperate async function and wrap it within here which will make it ugly :p

Reply:
Haha fair!